packages/api/src/routes/medical/internal-patient.ts

```diff
@@ -486,4 +489,31 @@ router.get(
   })
 );
 
+router.post(
+  "/mpi/block",
+  asyncHandler(async (req: Request, res: Response) => {
+    const dob = getFrom("query").orFail("dob", req);
+    const genderAtBirthRaw = getFrom("query").orFail("genderAtBirth", req);
+    if (genderAtBirthRaw !== "F" && genderAtBirthRaw !== "M") {
+      throw new Error("Invalid genderAtBirth value");
+    }
+    const genderAtBirth = genderAtBirthRaw as GenderAtBirth;
+    const firstNameInitial = getFrom("query").optional("firstNameInitial", req);
+    const lastNameInitial = getFrom("query").optional("lastNameInitial", req);
+
+    const blockedPatients = await blockPatients({
+      data: {
+        dob,
+        genderAtBirth,
+        firstNameInitial,
+        lastNameInitial,
+      },
+    });
+
```

Comment:
This endpoint was created because we didnt want to query the db directly from the lambda with code in core. That would have been a new pattern requiring a lot of refactoring. This isn't a perfect solution since it imposes some load on the app, but its a solid compromise since the patient matching operations are still offloaded, and we will be able to rate limit through the gateway and through mirth channels by limiting the number of threads

Reply:
Let's please add a documentation to the endpoint like we've been doing for others, and could briefly explain this there

also:
- there's no `block` business operation, this is a search/find
- this is not a MPI, let's avoid using this as we discussed yesterday

maybe `GET /internal/patient`?

Reply:
how about `GET /internal/patients` since we return multiple?

Reply:
Our standard is to use the singular form of nouns for endpoints, and we already have a `/internal/patient` - [link](https://github.com/metriport/metriport/blob/9aa0b9f6aea9dfa4114df6c3777f8d1c898401ee/packages/api/src/routes/internal.ts#L27)